3634/0/about_scoring_project.rb

```diff
@@ -0,0 +1,90 @@
+# rubocop:disable Metrics/AbcSize, Lint/UnneededCopDisableDirective
+require File.expand_path(File.dirname(__FILE__) + '/neo')
+
+# Greed is a dice game where you roll up to five dice to accumulate
+# points.  The following "score" function will be used to calculate the
+# score of a single roll of the dice.
+#
+# A greed roll is scored as follows:
+#
+# * A set of three ones is 1000 points
+#
+# * A set of three numbers (other than ones) is worth 100 times the
+#   number. (e.g. three fives is 500 points).
+#
+# * A one (that is not part of a set of three) is worth 100 points.
+#
+# * A five (that is not part of a set of three) is worth 50 points.
+#
+# * Everything else is worth 0 points.
+#
+#
+# Examples:
+#
+# score([1,1,1,5,1]) => 1150 points
+# score([2,3,4,6,2]) => 0 points
+# score([3,4,5,3,3]) => 350 points
+# score([1,5,1,2,4]) => 250 points
+#
+# More scoring examples are given in the tests below:
+#
+# Your goal is to write the score method.
+# :reek:FeatureEnvy
+# :reek:TooManyStatements
+def score(dice)
+  return raise DiceError, 'Use no more than 5 dices' unless dice.size < 6
+
+  points = 1000 * (dice.count(1) / 3)
+  points += 100 * (dice.count(1) % 3)
+  points += 50 * (dice.count(5) % 3)
+  points + (2..6).sum { |num| (100 * num) * (dice.count(num) / 3) } if dice != 1
```

Comment:
this line is overkill for case when there's no 2, 3, 4 or 6
imagine if you'll different shape object. 8 sides or 10 sides for example.
this will require to change range from (2..6) to new one.
think about how to refactor this
direct questions are appreciated